src/gcp/cloudfunctionsv2.ts

```diff
@@ -331,7 +331,7 @@ export async function createFunction(cloudFunction: InputCloudFunction): Promise
 
   cloudFunction.serviceConfig.environmentVariables = {
     ...cloudFunction.serviceConfig.environmentVariables,
-    FUNCTION_TARGET: functionId,
+    FUNCTION_TARGET: functionId.replace("-", "."),
```

Comment:
'replace' only replaces first instance of the search string. It should be using 'replaceAll'.